src/io/flutter/run/daemon/DeviceService.java

```diff
@@ -205,10 +205,11 @@ private DeviceDaemon chooseNextDaemon(Refreshable.Request<DeviceDaemon> request)
       return previous; // Don't do anything; current daemon is what we want.
     }
 
-    // Wait a bit to see if we get cancelled.
-    // This is to try to avoid starting a process only to immediately kill it.
+    // Wait a bit to see if we get cancelled. This is to try to avoid starting a process only to
+    // immediately kill it. Also, delay a bit in case the flutter tool just upgraded the sdk;
+    // we'll need a bit more time to start up.
     try {
-      Thread.sleep(50);
+      Thread.sleep(2000);
```

Comment:
I think this is related to the time to snapshot flutter_tools. So:
- lock the flutter lock
- test to update the sdk
- build a snapshot of flutter_tools
- unlock the flutter lock
- run the flutter_tools dart code

However, it looks like we don't lock on the mac (https://github.com/flutter/flutter/blob/master/bin/flutter#L74), which is likely the cause of this issue.